### ISSUE TITLE
question to "We are really terrible at digital colours, and digital photography."

### ISSUE THREAD

**Issue by piorekf** [Pomax/Pomax.github.io#28]
First of all thanks for fabulous article. I learned few new thinks thanks to you.
My question is: wouldn't this new, great way of storing wide colour range with six primaries require: new monitors (most of today's consumer monitors have even problems displaying full aRGB), changes in operating systems, changes in graphics programs, and probably even changes in graphics cards? Wouldn't this be the real reason why nobody will build a camera using such colour definitions?

**Comment by Pomax** [comment]
yes, and no. The improvements are maximised if there is a full chain. but graphics cards don't care: they don't work with colours, they're pure DSP boards and just process data "as instructed", so the problem is both cameras, and render devices, and as we're seeing with the development of 4K (which isn't just 'more pixels' but also a much wider range of colours) it is entirely possible to push for improvements. It's just a lot of effort to push for it. A 4K camera with a plain monitor is an improvement over a plain camera with a plain monitor, you have a lot more colour to work with before the final photograph (if you're a facebook poster, you don't care. If you're an amateur photographer, you care, even if it's only on one in a hundreds shots). And then when you can afford it and you buy a wider gamut monitor, you'll suddenly realise how much you care. A bit like when we went from 8 bit to 16 bit to 32 bit audio. A lot of people these days don't remember how amazing that change was (or never experienced it) but sound that seemed "just fine" because the playback was hissy and that's just what we expected were suddenly REALLY obviously defcient when 32 bit audio recording and playback made it into consumer devices, and the improvement in quality has fueled everything from webaudio to itunes. 

You don't need to upgrade the full stack all at the same time. You just need someone to go "look, we've got stuff that is much better than before, and everyone else is still stuck in the stone age, get with the effing program" and then have the power to force traction. Or, be a newcomer and demonstrate there's a LOT of money to be made in that space (like Blackmagic Design) so that competitors realise it's in their interested to invest in the same kind of technological advance.

**Comment by piorekf** [comment]
It's true that 4K standard shows that improvement is possible but this improvement is not done by users but by manufacturers and broadcasters who decided that they can do better. People are only buying the end product often even without knowing that video codecs have changed and that they can see more colours now.
When talking about choices consciously made by the users we should look at 10bit colours problem. And here the adoption is really poor. Macs, which are considered to be the platform for graphics, still can't support 10bit colours. Hardware that supports this is till very expensive (both graphic cards and monitors) and besides pros that do graphics for living not that many people can afford it.
Looking at all this I think that this would be the real killer of all those cool new things that you suggested and not only the reason that what we have now is "good enough".

**Comment by Pomax** [comment]
This feels like a much less important problem to me - higher contrast ratios are important, but even at 8 bits of intensity we can already at range-parity with reality, within reasonable limits (your monitor can't do true sunlight intensity, and that's great, because people would go blind).The fidelity may be too low to yield true smooth gradients to the human eye (so we're effectively looking at [intensity-posterized](https://en.wikipedia.org/wiki/Posterization) results) but we can intelligently use gamma control (either using a simple gamma function or more complex composite leveling) to make sure that, using the higher bit count source material, we map down to 8 bits in a way that at least still looks pleasing. Will it look seemless? Probably not, although it highly depends on the material rendered: natural scenes rarely suffer from a noticable lack of bits due to a lack of smooth gradients over long distances (and of course, if you do have a nature shot with a wide gradient, then yes, you'll notice), whereas art and product photography will run into this limitation much more often.

Still, there's only so many topics I can cover in a long read post, and between "we can already capture black and white, but the range between those could do with higher contrast fidelity" and "we *can't* capture all colours we can actually see, let alone debate whether we have gaps between the colours we get" I personally find the latter to be the bigger problem. I'd rather have parity with reality first, and then fix the contrast depth issue, than the other way around.

So, to invoke text from the article: "we suck at digital colour" and there isn't really one single thing that we can fix and then claim we're done - it's multiple problems that we ideally solve in lockstep with each other, but we probably won't. For now, I'll leave the contrast battle to others to fight, and I would be very interested in reading your article (if you write one) on topics I didn't surface in mine.

(and I may at some point get too frustrated with the lack of contrast depth in my own work, and end up writing a lament for that, too, some time in the future)
